utility/utils.py

```diff
@@ -2313,6 +2313,123 @@ def get_bucket_sync_status(node, bucket_name):
     return out
 
 
+def get_ceph_version_from_cluster(client):
+    """
+    Retrieve the Ceph version installed on a cluster using the provided client.
+
+    Args:
+        client : An instance of the client used for executing commands.
+
+    Returns:
+        str or None: The Ceph version if installed, or None if Ceph is not installed.
+
+    Raises:
+        ValueError: If the JSON output does not contain the expected version information.
+    """
+    out, rc = client.exec_command(
+        sudo=True,
+        cmd="ceph version -f json",
+        check_ec=False,
+    )
+    log.info(out)
+    ceph_version = json.loads(out)
+    version_string = ceph_version.get("version", None)
+    log.info(version_string)
+    if not version_string:
+        log.error("Ceph is not installed please install ceph")
+        return None
+    version_pattern = r"ceph version (\S+).*"
+    match = re.search(version_pattern, version_string)
+    re.search(version_pattern, version_string)
+    if not match:
+        raise RuntimeError("Failed to get ceph version from cluster")
+    ceph_version_installed = match.group(1)
+    return ceph_version_installed
+
+
+def get_ceph_version_from_repo(client, config):
+    """
+    Install Ceph and retrieve the version from the specified repository.
+
+    Args:
+        client (YourClientType): An instance of the client used for executing commands.
+        config (dict): Configuration parameters including 'rhbuild', 'base_url', and 'env_type'.
+
+    Returns:
+        str: The Ceph version retrieved from the repository.
+
+    Raises:
+        RuntimeError: If mandatory parameters are missing or the platform is unsupported.
+    """
+    podman_run = "podman run -it --rm"
+    rhel8_ubi_image = "registry.access.redhat.com/ubi8/ubi"
+    rhel9_ubi_image = "registry.access.redhat.com/ubi9/ubi"
+    exec_cmd = "sh -c"
+    yum_add_repo = "dnf config-manager --add-repo"
+    curl_add_repo = "curl -L -o /etc/yum.repos.d/upstream.repo"
+    ibm_license = (
+        "yum install --nogpgcheck -y ibm-storage-ceph-license && "
+        "touch /usr/share/ibm-storage-ceph-license/accept"
+    )
+    install_ceph_common = "yum install ceph-common -q --nogpgcheck -y"
```

Comment:
We are installing ceph-common tool from repo shared in run.py to get the veph version that would be installed.
Can it not be possible to get ceph version from repo path without installing  ceph-common on client? Because, what if the same client could be used for some testing where ceph-common with installed version was not required.
For eg., a case where repo ceph version is lower that ceph installed version on cluster, in this case, we have installed lower ceph version already to the client?
This concern is assuming the client used here, will also be used by test suites later on. If not, plz ignore.

Reply:
Hi Suma,
We are installing ceph-common inside a container in client node. This will not affect the current ceph-common which is installed on the client node.